static/js/services/changes.js

```diff
@@ -46,7 +46,7 @@ angular.module('inboxServices').factory('Changes',
       db.lastSeq = change.seq;
       Object.keys(db.callbacks).forEach(function(key) {
         var options = db.callbacks[key];
-        if (!options || !options.filter || options.filter(change)) {
+        if (options && options.filter && options.filter(change)) {
```

Comment:
This is not correct. This way if you don't give a filter your callback will never be called. The original logic called your callback if you didn't provide a filter, _or_ the filter returned `true`. This could be clearer, for example, by defaulting the filter to `function() { return true; }` on init.

Reply:
I don't think `options` should ever be falsey, because without a callback it's pointless to register a listener. This is one of those examples where `callback` shouldn't be part of the `options` map because it's not optional. Investigate why the `reports-content` listener is `null` - perhaps it's not being deregistered properly?